activerecord/lib/active_record/reflection.rb

```diff
@@ -199,7 +199,7 @@ def join_scopes(table, predicate_builder) # :nodoc:
       def klass_join_scope(table, predicate_builder) # :nodoc:
         if klass.current_scope
           klass.current_scope.clone.tap { |scope|
-            scope.joins_values = []
+            scope.joins_values = scope.left_outer_joins_values = []
```

Comment:
I'm sure it's fine, but can we throw a `.freeze` on there to make me feel better?